src/public/resolvers/queries/CorpusItem.ts

```diff
@@ -7,17 +7,18 @@ import { getCorpusItemFromApprovedItem } from '../../../shared/utils';
 import { UserInputError } from 'apollo-server-errors';
 
 /**
- * Pulls in approved corpus items for a given id.
+ * Pulls in approved corpus items for a given id or url.
  *
- * @param item
+ * @param item { id, url }
  * @param db
  */
-export async function getCorpusItem(item, { db }): Promise<CorpusItem> {
-  const { id } = item;
+export async function getCorpusItem({ id, url }, { db }): Promise<CorpusItem> {
+  const approvedItem = id
+    ? await getApprovedItemByExternalId(db, id)
+    : await getApprovedItemByUrl(db, url);
 
-  const approvedItem = await getApprovedItemByExternalId(db, id);
   if (!approvedItem) {
-    throw new UserInputError(`Could not find Corpus Item with ID of "${id}".`);
```

Comment:
We should not be throwing errors if things are not found in reference resolvers. In this case, it can return null and is not a UserInputError. its just non-existent.